File: lib/services/api_exception.dart
/// Excepción genérica para errores de comunicación con la API.
class ApiException implements Exception {
  final String message;
  final int? statusCode;

  const ApiException(this.message, {this.statusCode});

  @override
  String toString() =>
      'ApiException(statusCode: $statusCode, message: $message)';
}

File: backends/delivery-api/src/main/java/com/mycompany/delivery/api/util/ApiException.java
package com.mycompany.delivery.api.util;

/**
 * ✅ Excepción personalizada para manejar errores controlados dentro de la API.
 */
public class ApiException extends RuntimeException {

    private final int status;
    private final transient Object details;

    public ApiException(int status, String message) {
        super(message);
        this.status = status;
        this.details = null;
    }

    public ApiException(int status, String message, Throwable cause) {
        super(message, cause);
        this.status = status;
        this.details = null;
    }

    public ApiException(int status, String message, Object details) {
        super(message);
        this.status = status;
        this.details = details;
    }

    public int getStatus() { return status; }
    public Object getDetails() { return details; }
}
